### PR TITLE
feat: Adding github action to run the operator end-to-end tests.

### PR DESCRIPTION
**Pull request by lokeshrangineni**
* Adding github action to do the operator end-to-end testing. This action will create a kind cluster and runs the tests on it.

### CODE REVIEW

**Review by dmartinol — COMMENTED** [review]
lgtm

**Review by franciscojavierarceo — CHANGES_REQUESTED** [review]
Can we separate the actions upgrades from adding the operator? There have been side effects to the release process from upgrading stuff in the actions, so I'd like us to be able to revert or adjust easily without removing the operator.

**Comment by lokeshrangineni** [comment]
> Can we separate the actions upgrades from adding the operator? There have been side effects to the release process from upgrading stuff in the actions, so I'd like us to be able to revert or adjust easily without removing the operator.

I have separated out the warning fixes into separate [PR](https://github.com/feast-dev/feast/pull/4763)

**Review by dmartinol — COMMENTED** [review]
lgtm

**Review by franciscojavierarceo — APPROVED** [review]
Lgtm

File: .github/workflows/operator-e2e-integration-tests.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  operator-e2e-tests:
+    if:
+      ((github.event.action == 'labeled' && (github.event.label.name == 'approved' || github.event.label.name == 'lgtm' || github.event.label.name == 'ok-to-test')) ||
+      (github.event.action != 'labeled' && (contains(github.event.pull_request.labels.*.name, 'ok-to-test') || contains(github.event.pull_request.labels.*.name, 'approved') || contains(github.event.pull_request.labels.*.name, 'lgtm')))) &&
+      github.repository == 'feast-dev/feast'
+    runs-on: ubuntu-latest
+
+    services:
+      kind:
+        # Specify the Kubernetes version
```
[the remark this answers is not in the document]
Comment (lokeshrangineni) [reply .github/workflows/operator-e2e-integration-tests.yml:24]: I have revalidated again, each kind image version is mapped to the same kubernetes version.

**Pull request merged**